packages/firebase_analytics/firebase_analytics/lib/firebase_analytics.dart

```diff
@@ -24,14 +21,14 @@ class FirebaseAnalytics {
   ///
   ///     FirebaseAnalytics analytics = FirebaseAnalytics();
   ///     analytics.android?.setSessionTimeoutDuration(true);
-  final FirebaseAnalyticsAndroid android =
+  final FirebaseAnalyticsAndroid? android =
       defaultTargetPlatform == TargetPlatform.android && !kIsWeb
           ? FirebaseAnalyticsAndroid()
           : null;
 
   /// Logs a custom Flutter Analytics event with the given [name] and event [parameters].
   Future<void> logEvent(
-      {@required String name, Map<String, dynamic> parameters}) async {
+      {required String name, Map<String, Object>? parameters}) async {
```

Comment:
It appears that values in `parameters` should be nullable, according to the JS SDK

https://cs.opensource.google/firebase-sdk/firebase-js-sdk/+/master:packages/analytics-interop-types/index.d.ts;l=27?q=analytics%20logEvent&ss=firebase-sdk

So we'd have:

```suggestion
      {required String name, Map<String, Object?>? parameters}) async {
```